plugins/inputs/dpdk/dpdk.go

```diff
@@ -28,204 +30,272 @@ const (
+}
+
+func (dpdk *dpdk) getDpdkInMemorySocketPaths() []string {
+	filePaths := dpdk.socketGlobPath.Match()
+
+	var results []string
+	for _, filePath := range filePaths {
+		fileInfo, err := os.Stat(filePath)
+		if err != nil || fileInfo.IsDir() || !strings.Contains(filePath, dpdkSocketTemplateName) {
+			continue
+		}
+
+		if isInMemorySocketPath(filePath, dpdk.SocketPath) {
+			results = append(results, filePath)
+		}
+	}
+
+	return results
 }
 
 // Checks that user-supplied commands are unique and match DPDK commands format
-func (dpdk *dpdk) validateCommands() error {
+func (dpdk *dpdk) validateAdditionalCommands() error {
 	dpdk.AdditionalCommands = uniqueValues(dpdk.AdditionalCommands)
 
-	for _, commandWithParams := range dpdk.AdditionalCommands {
-		if len(commandWithParams) == 0 {
+	for _, cmd := range dpdk.AdditionalCommands {
+		if len(cmd) == 0 {
 			return fmt.Errorf("got empty command")
 		}
 
-		if commandWithParams[0] != '/' {
-			return fmt.Errorf("%q command should start with '/'", commandWithParams)
+		if cmd[0] != '/' {
+			return fmt.Errorf("%q command should start with '/'", cmd)
 		}
 
-		if commandWithoutParams := stripParams(commandWithParams); len(commandWithoutParams) >= maxCommandLength {
+		if commandWithoutParams := stripParams(cmd); len(commandWithoutParams) >= maxCommandLength {
 			return fmt.Errorf("%q command is too long. It shall be less than %v characters", commandWithoutParams, maxCommandLength)
 		}
 
-		if len(commandWithParams) >= maxCommandLengthWithParams {
-			return fmt.Errorf("command with parameters %q shall be less than %v characters", commandWithParams, maxCommandLengthWithParams)
+		if len(cmd) >= maxCommandLengthWithParams {
+			return fmt.Errorf("command with parameters %q shall be less than %v characters", cmd, maxCommandLengthWithParams)
 		}
 	}
 
 	return nil
 }
 
-// Gathers all unique commands and processes each command sequentially
-// Parallel processing could be achieved by running several instances of this plugin with different settings
-func (dpdk *dpdk) Gather(acc telegraf.Accumulator) error {
-	// This needs to be done during every `Gather(...)`, because DPDK can be restarted between consecutive
-	// `Gather(...)` cycles which can cause that it will be exposing different set of metrics.
-	commands := dpdk.gatherCommands(acc)
+// Establishes connections do DPDK telemetry sockets
+func (dpdk *dpdk) maintainConnections() error {
+	candidates := []string{dpdk.SocketPath}
+	if choice.Contains(dpdkPluginOptionInMemory, dpdk.PluginOptions) {
+		candidates = dpdk.getDpdkInMemorySocketPaths()
+	}
+
+	// Find sockets in the connected-sockets list that are not among
+	// the candidates anymore and thus need to be removed.
+	for i := 0; i < len(dpdk.connectors); i++ {
+		connector := dpdk.connectors[i]
+		var found bool
+		for _, candidate := range candidates {
+			if candidate == connector.pathToSocket {
+				found = true
+				break
+			}
+		}
+		if !found {
```

Comment:
This can be simplified to
```suggestion
		if !choice.Contains(connector.pathToSocket, candidates) {
```

Reply:
Addressed: https://github.com/influxdata/telegraf/pull/14308/commits/45611abfc77d44a5a877900ca2f20aa14fbbd485